### ISSUE TITLE
Cruise/Vessel Event Approval conditions

### ISSUE THREAD

**Issue by MarioCarloni** [WHOIGit/ooicgsn-roundabout#526]
Cruise Event approval should require all related Vessel Events approved.

**Comment by joannekoch** [comment]
I don't think this is working.  As shown below, the Vessel is not approved but the associated Cruise is showing that it is approved.

![Capture1](https://user-images.githubusercontent.com/65416034/158646177-2fef6ee9-5f38-4e54-9d8e-c5c01ce9f060.PNG)


![Capture](https://user-images.githubusercontent.com/65416034/158646664-38c0a140-ac5e-479b-a1aa-4b90fce6036d.PNG)

**Comment by joannekoch** [comment]
**Start with Cruise and related Vessel are not approved. This is working:** 
- Approve Vessel, approve related Cruise. Cruise "Review Requested" badge is removed from navtree for the Cruise.
- Do not approve Vessel, approve related Cruise. "Review Requested" badge remains on the navtree for the Cruise.

**Start with Cruise and related Vessel are approved. This is not working:**
- Unapprove Vessel, refresh the Cruise navtree. "Review Request" badge does not appear on the navtree for the Cruise which means that the Cruise is still in the approved state.

I was expecting the Cruise to be in the unapproved state when the the related Vessel is unapproved.

**Comment by MarioCarloni** [comment]
Hi @joannekoch, an Event must be Approved by two or more Users. The absence of the Review Requested badge is meant to represent only that the logged-in User is no longer required to Review an Event. Let me know if you have any other questions.

**Comment by joannekoch** [comment]
@MarioCarloni, I forgot about the two or more users..... Thanks for reminding me.

I don't think this is working as it should:
Logged in as the admin user, you can see that the vessel is only approved by one user, tech.
![Capture](https://user-images.githubusercontent.com/65416034/162029231-4c2c4033-3a99-4930-9da8-8d3b8950e712.PNG)


Logged in as admin, the cruise shows that it is in the approved state, but the vessel has not been approved by two users. Given that logic, the cruise should not be in the approved state. Is that Correct?

![Capture1](https://user-images.githubusercontent.com/65416034/162028768-5cf4e1a5-520a-4dd4-98c2-fd99500a1235.PNG)

**Comment by MarioCarloni** [comment]
Hi @joannekoch this is a great find! I'm seeing the same thing locally.

**Comment by joannekoch** [comment]
Hey @MarioCarloni I'm glad you could follow my explanation. Thanks for working on this!

**Comment by joannekoch** [comment]
Some scenarios are still not working as they should.

The cruise and vessel are approved by two users, but the cruise is in the "in progress" state when it should be in the "approved" state. To reproduce: login in as admin, toggle the vessel approved/unapproved. Refresh the cruise screen and watch the approved/in progress behavior on the cruise screen.

![Capture](https://user-images.githubusercontent.com/65416034/163185783-1b14e69a-6ef2-471a-91de-0081d95fc5e6.PNG)
![Capture1](https://user-images.githubusercontent.com/65416034/163185798-d421fbad-99a3-4daa-934e-804e55b41a7a.PNG)

Now unapproved the vessel as admin. Refresh the cruise screen. The cruise is still in the approved state.

![Capture](https://user-images.githubusercontent.com/65416034/163186362-991ebacb-7bd8-48a6-9972-909682c
![Capture1](https://user-images.githubusercontent.com/65416034/163186379-f644f88f-3f75-4698-9d94-e6ac440cc13c.PNG)
7c502.PNG)

**Comment by joannekoch** [comment]
Hey @MarioCarloni, after incorporating hotfix #545 I'm still seeing the behavior described above. If you approve the cruise and vessel as admin and tech users, and then unapprove the vessel as admin and refresh the cruise screen, the cruise is still in the approved state.
![Capture](https://user-images.githubusercontent.com/65416034/165153026-f5757e54-d209-4bb3-9fe5-b5fce5c32576.PNG)

If you unapprove the cruise as admin and then approve it, the "in progress" flag is not show.

![Capture](https://user-images.githubusercontent.com/65416034/165153327-5a76a4ff-991c-4e18-a534-c5b52a23629f.PNG)

If you then refresh the cruise browser window, the "in progress" flag appears as it should (because the vessel is not approved for the admin user).

![Capture](https://user-images.githubusercontent.com/65416034/165153582-36d5f915-499d-4b37-b21f-a105ff29e0a8.PNG)

**Comment by MarioCarloni** [comment]
@joannekoch I'm seeing some related UI glitches for Cruises. I had 2 Users Approve a Cruise Event. Upon An Approved badge did not appear on-Approval or page refresh, which was working before. However, there was an Event record of the Approval...

![image](https://user-images.githubusercontent.com/5469058/165334330-05786b71-3a52-4e22-b668-7fcd728d4c65.png)

**Comment by joannekoch** [comment]
This seems to be working much better now.
![Capture](https://user-images.githubusercontent.com/65416034/168827288-e7dbe54e-ced8-42b5-af55-c367b3045d58.PNG)

But, the "Review Requested" flag still appears in the navtree. It only goes away when I refresh the browser. 

@MarioCarloni - **Should the flag auto-refresh in the navtree?**

![Capture1](https://user-images.githubusercontent.com/65416034/168827311-834b60fd-876e-4144-97bd-c4e69e579499.PNG)

**Comment by MarioCarloni** [comment]
Hey @joannekoch the badge should refresh without page-reload. I'm seeing the behavior you're seeing on Testing.

**Comment by joannekoch** [comment]
The badges refresh without page-reload. This is working beautifully!
